tmdb_client.py

```diff
@@ -0,0 +1,61 @@
+import requests
+
+api_token = "api_token"
+
+
+def get_popular_movies():
+    endpoint = "https://api.themoviedb.org/3/movie/popular"
+    headers = {
+        "Authorization": f"Bearer {api_token}"
+    }
+    response = requests.get(endpoint, headers=headers)
+    return response.json()
+
+
+def get_poster_url(poster_api_path, size="w342"):
+    base_url = "https://image.tmdb.org/t/p/"
+    return f"{base_url}{size}/{poster_api_path}"
+
+
+def get_movies(how_many, list_type):
+    data = get_movies_list(list_type)
+    return data["results"][:how_many]
+
+
+def get_single_movie(movie_id):
+    endpoint = f"https://api.themoviedb.org/3/movie/{movie_id}"
```

Comment:
zauważ, że we wszystkich metodach poniżej masz zmienną `endpoint`, która w większości jest tym samym. Ma tylko inną końcówkę. Dlatego wyekstrahuj wspólną część do zmiennej, wyrzuć zmienną na górę pliku i twórz charaktrystyczne dla metod endpointy poprzez użycie tej zmiennej